autogluon/utils/tabular/ml/learner/default_learner.py

```diff
@@ -64,6 +67,15 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, scheduler_options=None, hy
             logger.log(20, f'Tuning Data Columns: {len(X_val.columns)}')
         time_preprocessing_start = time.time()
         logger.log(20, 'Preprocessing data ...')
+        if X_test_unlabeled is not None:
+            joint_feature_generator = copy.deepcopy(self.feature_generator)
+            # TODO: consider caching original data to file to save memory.
+            X_og = X.copy()
```

Comment:
Better to add save/load of pkl's for the data like is done in trainer. Should probably be done regardless of if unlabeled data is present.